frontend/Auth/FinishResetPassword.tsx

```diff
@@ -13,6 +13,10 @@ export function FinishResetPassword(props: RouterProps) {
     useState<FinishResetPasswordFormData>({
       password: "",
     });
+  const [tokenIsValid, setTokenIsValid] =
+    useState<boolean | undefined>(undefined);
```

Comment:
Very minor, but usually `undefined` indicates something that accidentally occurs as a result of the javascript programming language whereas `null` indicates that the developer intentionally wanted to not have a value. Since you intentionally wanted this to not have a value, using null is probably preferred. Same goes for line 19 with tokenIsExpired:

```suggestion
    useState<boolean | null>(null);
```

Reply:
Think over this more, I've realized that these variables are completely unused? They are unneeded since we have `tokenValidationResponse.tokenIsValid`

So I think these useStates can just be removed entirely